sklearn/linear_model/tests/test_bayes.py

```diff
@@ -288,3 +288,37 @@ def test_ard_regression_predict_normalize_true():
     clf = ARDRegression(normalize=True)
     clf.fit([[0, 0], [1, 1], [2, 2]], [0, 1, 2])
     clf.predict([[1, 1]], return_std=True)
+
+
+def test_dtype_match():
+    # Test that np.float32 input data is not cast to np.float64 when possible
+    X = np.array([[1, 1], [3, 4], [5, 7], [4, 1], [2, 6], [3, 10], [3, 2]])
+    y = np.array([1, 2, 3, 2, 0, 4, 5]).T
+    X_32 = np.array(X).astype(np.float32)
+    y_32 = np.array(y).astype(np.float32)
+    X_64 = np.array(X).astype(np.float64)
+    y_64 = np.array(y).astype(np.float64)
+
+    # check type consistency
+    for X, dtype in [(X_32, np.float32), (X_64, np.float64)]:
+        for y in [y_32, y_64]:
+            for model in [BayesianRidge(), ARDRegression()]:
```

Comment:
This can be parametrized. I'd see 1 parametrization over BayesianRidge and ARDRegression, and one parametrization over dtype (I don't think it's necessary to put different dtypes in for X and y).